src/shared/otterizecloud/otterizecloudclient/cloud_client.go

```diff
@@ -31,9 +33,13 @@ func NewClient(ctx context.Context) (graphql.Client, bool, error) {
 		AuthStyle:    oauth2.AuthStyleInParams,
 	}
 
-	tokenSrc := cfg.TokenSource(ctx)
-	graphqlUrl := fmt.Sprintf("%s/graphql/v1beta", apiAddress)
-	httpClient := oauth2.NewClient(ctx, tokenSrc)
+	// Timeout for oauth token acquisition is set using the http client passed to the token source context
+	clientWithTimeout := &http.Client{Timeout: clientTimeout}
+	ctxWithClient := context.WithValue(ctx, oauth2.HTTPClient, clientWithTimeout)
 
+	tokenSrc := cfg.TokenSource(ctxWithClient)
+	graphqlUrl := fmt.Sprintf("%s/graphql/v1beta", apiAddress)
+	httpClient := oauth2.NewClient(ctxWithClient, tokenSrc)
+	httpClient.Timeout = clientTimeout
```

Comment:
Sure this is required? doesn't it use the client you provided?

Reply:
Yes, they right it in the documentation, I'll add comment